shared/editor/rules/emoji.ts

```diff
@@ -1,12 +1,46 @@
-import MarkdownIt from "markdown-it";
+import MarkdownIt, { StateCore } from "markdown-it";
 import { full as emojiPlugin } from "markdown-it-emoji";
 import { nameToEmoji } from "../lib/emoji";
 
 type Options = MarkdownIt.Options & {
   emoji: boolean;
 };
 
+function customEmojiRule(state: StateCore) {
```

Comment:
RE: Lossy I don't think we need to have a way to import these from markdown at all, would remove this complexity